Add support of merging classes in withStyles

synaptiko opened the pull request:
Hello,

we just migrated to `tss-react` in MUIv4 based project and we found a small difference in how `withStyles` work in MUI and here.

The difference is in maybe a special case. Here's a small example:
```tsx
import InputBase from '@material-ui/core/InputBase';
import { withStyles } from 'common-ui/styles';

export const CustomInputBase = withStyles(
  InputBase,
  (theme) =>
    ({
      input: {
        backgroundColor: theme.palette.grey[50],
      },
    } as const)
);

const StyledInput = withStyles(CustomInputBase, () => ({
  input: {
     backgroundColor: 'red'
  },
}));
```

The problem is that `StyledInput` won't get `red` color as one would expect. The reason is that `withStyles` currently doesn't merge `classes` passed in `props` of the wrapping component.

I tried to add it by using `mergeClasses` function where I noticed another small issue. The problem is when `classes` doesn't contain all the keys; then the key present on `classesOv` would be missing in the output classes object.

garronej commented:
Hi, thank you very much for that PR.  
Sorry I didn't manage to make time to test it yet although it looks good!  
I published a beta version incorporating your changes so you are not blocked until it make it to main. 
[3.3.2-beta.0](https://www.npmjs.com/package/tss-react/v/3.3.2-beta.0).

synaptiko commented:
@garronej Thanks for the quick reaction, I just upgraded to the beta version and it seems to work for our use-case.

garronej commented:
I'm testing the changes right now.  
I happy you're choosing to use TSS not just only out of necessity to upgrade to MUIv5. 😄

garronej commented:
Hi @synaptiko,  
Good catch on this bug, it was a pretty big one.  
I got confused at first because the sniped you provided works as expected (if it's not the case for you, you might have a problem with your configuration).  
Anyway I wrote another [test](https://github.com/garronej/tss-react/blob/37dad3ee1b4356b47e783e54d807f3c038a5c32b/src/test/apps/muiV4ssr/pages/index.tsx#L349-L376) and polished a little bit your PR that was otherwise very good.  

Thanks for your contribution,

synaptiko commented:
> I got confused at first because the sniped you provided works as expected

I extracted it from our code and simplified it for this PR but I think I did a mistake there.

Thanks for the test, I wanted to provide it too but wasn't sure what's the right process.

garronej commented:
No problem, you submitted a clean PR when you could just have opened an issue.
